### DISCUSSION TITLE
Umbraco.Forms.Themes GitHub repository

### DISCUSSION THREAD

**Discussion by stephen-sherman** [umbraco/Umbraco.Forms.Issues#1524] [Ideas]
One of the most common frustrations when upgrading Umbraco Forms to a new version is figuring out what changes need to be applied to a custom theme.

Umbraco Forms' default themes are currently provided as .zip files in [the Umbraco Forms documentation](https://docs.umbraco.com/umbraco-forms/developer/themes) - but these aren't readily available for every version, and don't show what has changed compared to a given previous version.

Old default versions are often not available, and (depending on the level of modification) comparing a custom theme to the latest default version can be time consuming.

A public Umbraco.Forms.Themes GitHub repository would be a useful tool for providing access to new and historical theme packages, and allowing developers to more easily see the changes between different versions.

**Comment by AndyButland** [comment]
Could well be a very good idea, but just to clarify a couple of points for what we are doing at the moment:

- We don't release the zip files for every version, but we do when they are any changes.  So if for example there's a version for 14.0.1, not 14.0.2 and is for 14.1.0, then that just means the only changes came between 14.0.1 and 14.1,0.
- Old versions are available on our from the "Source Code" tab of the [Forms package page](https://our.umbraco.com/packages/developer-tools/umbraco-forms/).

**Comment by stephen-sherman** [comment]
Argh, sorry Andy - I completely overlooked the part of the documentation that mentions where the older versions can be found...

**Comment by skttl** [comment]
Still, a github repository would make it much easier to see how the default theme does. Right now I have to download the theme, unzip it, open it in some editor etc. just to be able to cross reference with my own theme.

Would also be easier to see differences between versions.

You could always disable issues and pull requests on the repository.

**Comment by ronaldbarendse** [comment]
I've just updated the documentation to list all Forms versions that contain changes to the default theme (per major), so you can now directly download the ZIP file you need from this page. However, as mentioned above, not every Forms version will have changes, so you'll have to pick the closest lower version (or simply update Forms to the latest).

I can see the benefit of having separate public Umbraco.Forms.Themes repository, especially because that allows comparing changes between tagged versions and we can even have release notes that summarize the changes. Disabling Pull Requests on a GitHub repository isn't possible though 🙃 Ideally, we'd like to completely remove these extra manual steps and have a built-in way to 'override' specific built-in views within the CMS, as mentioned in https://github.com/umbraco/Umbraco.Forms.Issues/issues/1162.

**Comment by umbracocommunity** [comment]
This issue has been mentioned on **Umbraco community forum**. There might be relevant details there:

https://forum.umbraco.com/t/customizing-umbraco-forms-themes-css-js-from-default-theme/6743/6

**Comment by warrenbuckley** [comment]
## The above Forum Post was linked here ⏫
To give some context - it was because I was trying to find and see the default files, including Views, CSS & JS that Umbraco Form uses.

I ended up asking on the forum on where I could find the CSS & JS bits for the theme but I resolved it by doing a `dotnet publish` in order to see the Razor Views and the JS and CSS files that get put onto the page when using Umbraco Forms.

It would be good to add another +1 here to this issue to consider a way on distributing the theme files for customisation in a lot easier way.

**Comment by skttl** [comment]
> Disabling Pull Requests on a GitHub repository isn't possible though

Get a bot to auto reject any pull request?
